src/efftester.ml

```diff
@@ -290,275 +304,328 @@ exception No_solution
-	                  else removeMultiMap polyentry var rEnv)
-	  returnEnv (returnTypes (normalize_eff s))
-      | None ->
-        returnEnv in
-    List.fold_left (fun rEnv rt ->
-                      if ftv rt = [] (* return type polymorphic? Then syntactic comp. will not find it *)
-		      then addMultiMap rt var rEnv
-		      else addMultiMap polyentry var rEnv) fixedReturnEnv rts in
-  (env', revEnv', returnEnv')
+        List.fold_left
+          (fun rEnv rt ->
+            if ftv rt = []
+               (* return type polymorphic? Then syntactic comp. will not find it *)
+            then removeMultiMap rt var rEnv
+            else removeMultiMap polyentry var rEnv)
+          returnEnv
+          (returnTypes (normalize_eff s))
+      | None -> returnEnv
+    in
+    List.fold_left
+      (fun rEnv rt ->
+        if ftv rt = []
+           (* return type polymorphic? Then syntactic comp. will not find it *)
+        then addMultiMap rt var rEnv
+        else addMultiMap polyentry var rEnv)
+      fixedReturnEnv
+      rts
+  in
+  env', revEnv', returnEnv'
+;;
 
 let lookupVar x (env, _, _) =
-  try Some (VarMap.find x env) with Not_found -> None
+  try Some (VarMap.find x env) with
+  | Not_found -> None
+;;
 
 let lookupType s (_, revEnv, _) =
-  try TypeMap.find s revEnv
-  with Not_found -> VarSet.empty
+  try TypeMap.find s revEnv with
+  | Not_found -> VarSet.empty
+;;
 
 let lookupReturn s (env, _, returnEnv) =
-  let concreteSet = try TypeMap.find s returnEnv
-                    with Not_found -> VarSet.empty in
+  let concreteSet =
+    try TypeMap.find s returnEnv with
+    | Not_found -> VarSet.empty
+  in
   let arity_s = arity s in
   let rec has_compat_rt t =
-    (arity t = arity_s && types_compat t s) || (match t with
-      | Fun (_,_,rt) -> has_compat_rt rt
-      | _            -> false) in
+    (arity t = arity_s && types_compat t s)
+    ||
+    match t with
+    | Fun (_, _, rt) -> has_compat_rt rt
+    | _ -> false
+  in
   let polySet =
-    VarSet.fold (fun x acc -> if has_compat_rt (VarMap.find x env) then VarSet.add x acc else acc)
-      (try TypeMap.find polyentry returnEnv with Not_found -> VarSet.empty) VarSet.empty in
+    VarSet.fold
+      (fun x acc -> if has_compat_rt (VarMap.find x env) then VarSet.add x acc else acc)
+      (try TypeMap.find polyentry returnEnv with
+      | Not_found -> VarSet.empty)
+      VarSet.empty
+  in
   VarSet.union concreteSet polySet
-
+;;
 
 (** Generators *)
 
 let a_code = int_of_char 'a'
 let z_code = int_of_char 'z'
 let alphaGen = Gen.map char_of_int (Gen.int_range a_code z_code)
 let varGen = Gen.map (String.make 1) alphaGen
-
 let stringGen = Gen.small_string ~gen:alphaGen
 let stringToString s = "\"" ^ s ^ "\""
-
 let sqrt i = int_of_float (sqrt (float_of_int i))
 
 let arb_int =
-  frequency
-    [(10,small_int);
-     (5,int);
-     (1, oneofl [min_int;-1;0;1;max_int])]
-  
+  frequency [ 10, small_int; 5, int; 1, oneofl [ min_int; -1; 0; 1; max_int ] ]
```

Comment:
This would be more readable with one item per line, but I haven't found a way to force `ocamlformat` to do it. (The two modes of `break-collection-expressions` are `wrap` or `fit-or-vertical`, and none forces systematic breaking; in particular even using a local `[@@ocamlformat]` attribute does not help). On the other hand, even the non-broken list is sensibly more readable with `--parens-tuple=always`, which I would therefore recommend as the default.